routers/api/actions/artifacts.go

```diff
@@ -192,6 +194,14 @@ func (ar artifactRoutes) getUploadArtifactURL(ctx *ArtifactContext) {
 		return
 	}
 
+	// set retention days
+	if req.RetentionDays > 0 {
+		cacheKey := fmt.Sprintf("actions_artifact_retention_days_%d_%d", task.ID, runID)
+		if err := cache.GetCache().Put(cacheKey, req.RetentionDays, 0); err != nil {
```

Comment:
Why we need to use cache? And cache maybe disabled?

Reply:
It receives `RenteionDays` in previous upload request. I need a method to pass the param to the following requests. If cache disabled, Should use memory map ?

Reply:
Hm, could we store it on database?

Reply:
I ever suggested to force to enable "cache" (builtin as fallback)

Without cache, many functions do not work correctly.

Reply:
But, I do not agree with the reason "I need a method to pass the param to the following requests."

It's an abuse of the cache.

Reply:
> Hm, could we store it on database?

In current design, one artifact db record  is created after getting `RetentionDays` params. So need a new temp table to save the value ?

Reply:
> I ever suggested to force to enable "cache" (builtin as fallback). Without cache, many functions do not work correctly.
> 
> But, I do not agree with the reason "I need a method to pass the param to the following requests."
> 
> It's an abuse of the cache, and it would cause various problems, for example: the cache items could be evicted at any time.

I think Session is better. Any case use Session in other module ?

Reply:
API doesn't support session now.

Reply:
Now use query string to pass `retention-days` to next uploading requests